### PR TITLE
Remove "Estimated" option from the deposit list advanced filters

### LINKED ISSUE #7630: Remove "Estimated" advanced filter from the deposits list page
We should remove the `Estimated` deposit status option from the deposits list advanced filters since we will no longer be including `Estimated` deposits in this view (#7604).

estimated-filter

### PR DESCRIPTION
**Pull request by Jinksi**
Fixes #7630

#### Changes proposed in this Pull Request

This PR removes the `Estimated` option from the advanced filters on the deposits list.

Along with the UI change, the removal of the `estimated` value from the `DepositStatus` TS type was needed. This results in many code changes unrelated to the focus of this PR.

**Before/After**

image
image

**Notes**
* This PR doesn't remove the `Estimated` deposits from this list, handled in #7604.
* This PR doesn't remove the `Estimated` deposits from the exported CSV, handled in #7631.
* This PR breaks the **Payments → Overview → Next deposit** UI by deleting the `Estimated` text from the `DepositStatusChip`. This will be handled in a new issue replacing this UI TBC.
	image

#### Testing instructions

* Visit the **Payments → Deposits** screen.
* Select **Advanced Filters** at the top of the table.
* **Add a Filter** and select **Status**.
* Ensure `Estimated` is not an available option.
* Ensure filters work as expected.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7632` or branch name `fix/7630-remove-estimated-deposit-status-filter-option` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/7630-remove-estimated-deposit-status-filter-option) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: dafdcf6018b321bc6bb8aa142ef36a5307ccaf58
- Build time: 2023-11-14 03:17:12 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/components/account-balances/test/index.test.tsx [hunk 1]
```diff
@@ -138,19 +138,7 @@ const createMockOverview = (
...
-			automatic: true,
-			currency: null,
-			bankAccount: null,
-			created: Date.now(),
-			date: Date.now(),
-			fee: 0,
-			fee_percentage: 0,
-			status: 'estimated',
-		},
+		nextScheduled: undefined,
```
Comment (Jinksi) [inline client/components/account-balances/test/index.test.tsx:141]: `nextScheduled` wasn't used for this test suite, made apparent by the lack of additional changes required in this or related test files.

File: client/components/deposits-overview/utils.ts [hunk 2]
```diff
@@ -29,7 +29,7 @@ export const getNextDeposit = (
 		return {
```
Comment (Jinksi) [inline client/components/deposits-overview/utils.ts:29]: Note that this entire file could be inlined into the only consumer of the `getNextDeposit` function: the `NextDepositDetails` component.

https://github.com/Automattic/woocommerce-payments/blob/d1384606930f7dcc2013804c0e51965e2cafc6f4/client/components/deposits-overview/next-deposit.tsx#L123

Most of this code will likely be removed when we redo the **Payments → Overview** UI as part of this project: issue TBC.

File: client/deposits/strings.ts [hunk 3]
```diff
@@ -5,16 +5,21 @@
...
+
+import type { DepositStatus } from 'wcpay/types/deposits';
+
 export const displayType = {
 	deposit: __( 'Deposit', 'woocommerce-payments' ),
 	withdrawal: __( 'Withdrawal', 'woocommerce-payments' ),
 };
 
-export const displayStatus = {
+export const displayStatus: Record< DepositStatus, string > = {
```
Comment (Jinksi) [inline client/deposits/strings.ts:19]: I've explicitly linked this to `DepositStatus` to help communicate the intended purpose of this mapping and help prevent potential bugs that may occur if we leave out a deposit status value.

File: client/components/account-balances/test/index.test.tsx (hunk 1 above)
Comment (haszari) [reply client/components/account-balances/test/index.test.tsx:141]: Assuming we can't delete the `nextScheduled` key from this test data for some reason?

File: client/components/deposit-status-chip/test/index.test.tsx [hunk 4]
```diff
@@ -10,13 +10,6 @@ import { render } from '@testing-library/react';
...
 
 describe( 'Deposits status chip renders', () => {
-	test( 'Renders In Transit status chip.', () => {
-		const { getByText } = render(
-			<DepositStatusChip status="estimated" />
-		);
-		expect( getByText( 'Estimated' ) ).toBeTruthy();
-	} );
-
 	test( 'Renders In Transit status chip.', () => {
```
Comment (haszari) [inline client/components/deposit-status-chip/test/index.test.tsx]: Let's give these more accurate test names while we're here, I'm assuming each one tests a specific status. E.g. `Renders pending status chip` (or whatever it really means to merchants).

File: client/components/deposits-overview/test/index.tsx [hunk 5]
```diff
@@ -550,7 +550,7 @@ describe( 'Paused Deposit notice Renders', () => {
 		);
 	} );
 	test( 'When available balance is positive', () => {
-		const overview = createMockOverview( 'usd', 100, 0, 'estimated' );
+		const overview = createMockOverview( 'usd', 100, 0, 'pending' );
```
Comment (haszari) [inline client/components/deposits-overview/test/index.tsx:553]: Woah, lots of tests that care about `estimated`!

File: client/components/deposits-overview/utils.ts (hunk 2 above)
Comment (haszari) [reply client/components/deposits-overview/utils.ts:29]: Sounds like a great refactor. Side note, if we were to keep it, `pending` seems like a funny default, the most common would be `paid`.

File: client/components/deposits-overview/utils.ts [hunk 6]
```diff
@@ -39,7 +39,7 @@ export const getNextDeposit = (
 	return {
 		id: nextScheduled.id,
 		date: nextScheduled.date ?? 0,
-		status: nextScheduled.status ?? 'estimated',
+		status: nextScheduled.status ?? '-',
 		amount: formatCurrency( nextScheduled.amount ?? 0, currency ),
```
Comment (haszari) [inline client/components/deposits-overview/utils.ts:43]: I'm a little confused by these changes – `getNextDeposit()` returns `status: -`, what does this mean? Is the code dead? If it returns `estimated` what breaks?

I guess this code is becoming redundant; `getNextDeposit` seems coupled to the idea of estimating/predicting deposits.

File: client/data/transactions/hooks.ts [hunk 7]
```diff
@@ -25,13 +26,7 @@ export interface Transaction {
...
 	customer_currency: string;
 	deposit_id?: string;
-	deposit_status?:
-		| 'paid'
-		| 'pending'
-		| 'in_transit'
-		| 'canceled'
-		| 'failed'
-		| 'estimated';
+	deposit_status?: DepositStatus;
```
Comment (haszari) [inline client/data/transactions/hooks.ts:29]: Nice :)

**Review by haszari — APPROVED** [review]
Looks good to me – though I haven't tested. 

Are we pulling a thread that we need to keep pulling? `getNextDeposit` / utils.ts seems like it's half way there.

No blockers, though possibly this needs more testing before merge (I'm happy to do some testing). 

Also we need to consider what happens if we merge UI PRs before code 6.8 freeze. I'm _hoping_ we can ship incrementally but we need to think that through and decide as a team.

File: client/components/deposit-status-chip/test/index.test.tsx (hunk 4 above)
Comment (Jinksi) [reply client/components/deposit-status-chip/test/index.test.tsx]: Great catch: updated in bea8783.

**Pull request merged**